blocks/api/usage.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Internal dependencies
+ */
+import { getBlockType, getBlockTypes } from './registration';
+
+let recentBlockNames = [];
+const maxRecent = 8;
+
+// TODO: replace storage of recent blocks in this module with calls to an API
+export function getRecent() {
+	if ( 0 === recentBlockNames.length ) {
+		recentBlockNames = getBlockTypes()
+			.filter( blockType => 'common' === blockType.category )
+			.map( blockType => blockType.name );
+	}
+	return recentBlockNames.map( getBlockType );
+}
+
+export function recordUsage( blockType ) {
+	recentBlockNames = [
+		blockType.name,
+		...recentBlockNames.filter( name => name !== blockType.name ),
+	].slice( 0, maxRecent );
+}
```

Comment:
I was going to, but they will be gone in the next PR, which stores and calculates these on the server side, and is unit tested